### PR TITLE
Demo 2502 add mysql integration v2

### PR DESCRIPTION
**Pull request by moonlight-komorebi**
Second attempt at PR. 
Primary changes are in the last two commits.

### CODE REVIEW

File: deploy/databases/mysql/linux/roles/onafterstart/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    mysql_port: "{{ database_port | default(3306) }}"
+
+- name: Generate newrelic user password
+  set_fact:
+    newrelic_user_password: "{{ lookup('password', '/dev/null length=15 chars=ascii_letters') }}"
+
+- name: Create db user for newrelic mysql integration
+  community.mysql.mysql_user:
+    login_user: "root"
+    login_password: "{{ root_user_password }}"
```
Comment (moonlight-komorebi) [inline deploy/databases/mysql/linux/roles/onafterstart/tasks/main.yml]: change this to {{ database_root_password }}

File: deploy/databases/mysql/linux/roles/onafterstart/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+- name: Ensure newrelic_license_key is defined
+  fail:
+    msg: "newrelic_license_key is required for configuring newrelic infrastructure agent"
+  when: newrelic_license_key is not defined
+
+- name: Ensure root_user_password is defined
+  fail:
+    msg: "root_user_password is required for creating the integration's database user"
+  when: root_user_password is not defined
```
Comment (moonlight-komorebi) [inline deploy/databases/mysql/linux/roles/onafterstart/tasks/main.yml]: change this to {{ database_root_password }}

File: deploy/databases/mysql/linux/roles/instrument/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
+
+- fail:
+    msg: "version is required for preparing NewRelic MySQL Integration"
+  when: agent_version is not defined
+
```
Comment (moonlight-komorebi) [inline deploy/databases/mysql/linux/roles/instrument/tasks/main.yml]: add a comment explaining the loop

**Review by gabeobrien — APPROVED** [review]
Works, and look good.

**Pull request merged**